tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -197,3 +202,200 @@ async def with_ci_connector_ops(context: ConnectorTestContext) -> Container:
+) -> Container:
+    """Bind a container to a docker host. If the docker-cli is installed in this container it will use the dockerd service as a docker host.
+
+    Args:
+        context (ConnectorTestContext): The current connector test context.
+        container (Container): The container to bind to the docker host.
+        shared_volume (Optional, optional): A tuple in the form of (mounted path, cache volume) that will be both mounted to the container and the dockerd container. Defaults to None.
+        docker_cache_volume_name (Optional[str], optional): The name of the docker cache volume. Changing the volume name provides docker context isolation. Defaults to None.
+
+    Returns:
+        Container: The container bound to the docker host.
+    """
+    dockerd = with_dockerd_service(context, shared_volume, docker_cache_volume_name)
+    bound = container.with_env_variable("DOCKER_HOST", "tcp://docker:2375").with_service_binding("docker", dockerd)
+    if shared_volume:
+        bound = bound.with_mounted_cache(*shared_volume)
+    return bound
+
+
+def with_docker_cli(
+    context: ConnectorTestContext, shared_volume: Optional(Tuple[str, CacheVolume]) = None, docker_cache_volume_name: Optional[str] = None
+) -> Container:
+    """Create a container with the docker CLI installed and bound to a persistent docker host.
+
+    Args:
+        context (ConnectorTestContext): The current connector test context.
+        shared_volume (Optional, optional): A tuple in the form of (mounted path, cache volume) that will be both mounted to the container and the dockerd container. Defaults to None.
+        docker_cache_volume_name (Optional[str], optional): The name of the docker cache volume. Changing the volume name provides docker context isolation. Defaults to None.
+
+    Returns:
+        Container: A docker cli container bound to a docker host.
+    """
+    docker_cli = context.dagger_client.container().from_("docker:23.0.1-cli")
+    return with_bound_docker_host(context, docker_cli, shared_volume, docker_cache_volume_name)
+
+
+def with_connector_acceptance_test(context: ConnectorTestContext) -> Container:
+    """Create a container to run connector acceptance tests, bound to a persistent docker host.
+
+    Args:
+        context (ConnectorTestContext): The current connector test context.
+
+    Returns:
+        Container: A container with connector acceptance tests installed.
+    """
+    if context.connector_acceptance_test_image.endswith(":dev"):
+        cat_container = context.connector_acceptance_test_source_dir.docker_build()
+    else:
+        cat_container = context.dagger_client.container().from_(context.connector_acceptance_test_image)
+    shared_tmp_volume = ("/tmp", context.dagger_client.cache_volume("share-tmp-cat"))
+    return (
+        with_bound_docker_host(context, cat_container, shared_tmp_volume)
+        .with_entrypoint(["pip"])
+        .with_exec(["install", "pytest-custom_exit_code"])
```

Comment:
Simpler to use a single exec rather than combining with entrypoint if the entrypoint isn't meant to be leveraged later:

```python
        .with_exec(["pip", "install", "pytest-custom_exit_code"])
```
If you need to clear an entrypoint the image may have:
```python
        .with_entrypoint([])
```